# XGBoost
Let's load iris database and train model with 80% of total database and then test the built model using rest of 20% of data.

In [1]:
from sklearn.datasets import load_iris

In [2]:
iris = load_iris()
print(iris.data.shape)
print(iris.target_names)

(150, 4)
['setosa' 'versicolor' 'virginica']


iris database is having 150 rows with four features and three species

In [3]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(iris.data,iris.target,test_size=0.2,random_state=0)

Let's load the xgboost and covert the data into Dmatrix which xgboost expects.

In [4]:
import xgboost as xgb

C:\Users\C5225062\AppData\Local\Continuum\anaconda3\lib\site-packages\dask\config.py:168: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  data = yaml.load(f.read()) or {}
C:\Users\C5225062\AppData\Local\Continuum\anaconda3\lib\site-packages\distributed\config.py:20: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  defaults = yaml.load(f)


In [5]:
train = xgb.DMatrix(X_train, label=y_train)
test = xgb.DMatrix(X_test, label=y_test)

Now we will define hyper parameters to build model, We are choosing softmax as this multiclass classification.

In [6]:
param = {
    'max_depth':4,#Max leaf node depth
    'eta':0.3,#similar to learning rate
    'objective':'multi:softmax',#activation function
    'num_class':3
}
epochs=10

In [7]:
model = xgb.train(param,train,epochs)

In [8]:
prediction = model.predict(test)
print(prediction)

[2. 1. 0. 2. 0. 2. 0. 1. 1. 1. 2. 1. 1. 1. 1. 0. 1. 1. 0. 0. 2. 1. 0. 0.
 2. 0. 0. 1. 1. 0.]


Let's measure accuracy

In [9]:
from sklearn.metrics import accuracy_score
print(accuracy_score(prediction,y_test))

1.0


That's awesome. XGBoost is having high computation power and accuracy!!